youtube_dl/extractor/rai.py

```diff
@@ -55,181 +55,157 @@ def _extract_relinker_formats(self, relinker_url, video_id):
+        'md5': '340aa3b7afb54bfd14a8c11786450d76',
+        'info_dict': {
+            'id': 'e06118bb-59a9-4636-b914-498e4cfd2c66',
+            'ext': 'mp4',
+            'title': 'La Casa Bianca',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': r're:^Rai.+',
+            'description': 're:^[A-Za-z]+'
+        }
+    }, {
+        'url': 'http://www.raiplay.it/video/2016/11/gazebotraindesi-efebe701-969c-4593-92f3-285f0d1ce750.html?',
+        'md5': 'ed4da3d70ccf8129a33ab16b34d20ab8',
+        'info_dict': {
+            'id': 'efebe701-969c-4593-92f3-285f0d1ce750',
+            'ext': 'mp4',
+            'title': 'Gazebo - #gazebotraindesi',
+            'thumbnail': r're:^https?://.*\.png$',
+            'uploader': r're:^Rai.+',
+            'description': r're:^[A-Za-z]+'
+        }
+    }, {
+        'url': 'http://www.raiplay.it/video/2014/04/Report-del-07042014-cb27157f-9dd0-4aee-b788-b1f67643a391.html',
+        'md5': '8970abf8caf8aef4696e7b1f2adfc696',
+        'info_dict': {
+            'id': 'cb27157f-9dd0-4aee-b788-b1f67643a391',
+            'ext': 'mp4',
+            'title': 'Report - Report del 07/04/2014',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': r're:^Rai.+',
+            'description': r're:^[A-Za-z]+'
+        }
+    }]
+    _RESOLUTION = '600x400'
 
-        formats = []
-        media_type = media['type']
-        if 'Audio' in media_type:
-            formats.append({
-                'format_id': media.get('formatoAudio'),
-                'url': media['audioUrl'],
-                'ext': media.get('formatoAudio'),
-            })
-        elif 'Video' in media_type:
-            formats.extend(self._extract_relinker_formats(media['mediaUri'], content_id))
-            self._sort_formats(formats)
-        else:
-            raise ExtractorError('not a media file')
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
 
-        subtitles = {}
-        captions = media.get('subtitlesUrl')
-        if captions:
-            STL_EXT = '.stl'
-            SRT_EXT = '.srt'
-            if captions.endswith(STL_EXT):
-                captions = captions[:-len(STL_EXT)] + SRT_EXT
-            subtitles['it'] = [{
-                'ext': 'srt',
-                'url': captions,
-            }]
+        webpage = self._download_webpage(url, video_id)
 
-        return {
-            'id': content_id,
-            'title': media['name'],
-            'description': media.get('desc'),
-            'thumbnails': thumbnails,
-            'uploader': media.get('author'),
-            'upload_date': unified_strdate(media.get('date')),
-            'duration': parse_duration(media.get('length')),
-            'formats': formats,
-            'subtitles': subtitles,
-        }
+        canonical_url = self._og_search_url(webpage)
```

Comment:
Everything apart from query is the same and this unnecessary webpage downloading should be removed.